### PR TITLE
#138 table of contents (C++/C#/Go/Haskell)

### PR DESCRIPTION
**Pull request by iskandarzulkarnaien**
Adds table of contents (toc) to articles without one and hides them except for mobile and print view as discussed in #138 . 

This PR affects the below chapters under `Programming Languages`. Kindly use the below checklist for reviewing.

C++:
- [ ] Web view (Expected: in-article toc absent)
- [ ] Mobile view (Expected: in-article toc present, right nav panel absent)
- [ ] Print view (Expected: in-article toc present, left and right nav panels absent)

C#:
- [ ] Web view (Expected: in-article toc absent)
- [ ] Mobile view (Expected: in-article toc present, right nav panel absent)
- [ ] Print view (Expected: in-article toc present, left and right nav panels absent)

Go:
- [ ] Web view (Expected: in-article toc absent)
- [ ] Mobile view (Expected: in-article toc present, right nav panel absent)
- [ ] Print view (Expected: in-article toc present, left and right nav panels absent)

Haskell: 
- [ ] Web view (Expected: in-article toc absent)
- [ ] Mobile view (Expected: in-article toc present, right nav panel absent)
- [ ] Print view (Expected: in-article toc present, left and right nav panels absent)

Devtools shortcuts
Firefox:
<kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>M</kbd>

Chrome:
<kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>I</kbd>

### CODE REVIEW

**Comment by jamessspanggg** [comment]
@iskandarzulkarnaien the netlify build failed. Could you retrigger/check what happened with the build?

**Comment by iskandarzulkarnaien** [comment]
> 
> 
> @iskandarzulkarnaien the netlify build failed. Could you retrigger/check what happened with the build?

Ah thats unfortunate. I'm working on something else at the moment but I'll let you and Brian know when I've resolved the issue

**Comment by iskandarzulkarnaien** [comment]
@jamessspanggg @nbriannl The deploy preview is up. Not sure what went wrong yesterday. Thanks for the review help!

File: css/main.css [hunk 1]
```diff
@@ -27,3 +27,14 @@ mark {
 .indented-level3 {
     padding-left: 60px;
 }
+
+/* Responsive table of contents (toc) */
+
+/* Hides toc on screen widths smaller than 1300px */
+/* page-nav appears on screens 1300px wide and above */
+@media screen and (min-width: 1299.98px) {
```
Comment (jamessspanggg) [inline css/main.css:35]: Just one question, why not just 1300px?

**Review by jamessspanggg — APPROVED** [review]
LGTM! The responsive TOC works fine.

**Review by nbriannl — APPROVED** [review]
LGTM. Thanks for tidying things up.

File: css/main.css (hunk 1 above)
Comment (iskandarzulkarnaien) [reply css/main.css:35]: I wanted to be consistent with `page-nav.css` which has the following:

```
/* Responsive page navigation */

/* Hides page navigation on screen widths smaller than 1300px */
@media screen and (max-width: 1299.98px) {
    #page-nav {
        border: none;
        overflow: hidden;
        padding: 0;
        width: 0;
    }
}
```

**Comment by iskandarzulkarnaien** [comment]
Thanks for the help @jamessspanggg and @nbriannl !

@damithc my PR is ready to be merged

**Pull request merged**